src/tigris.ts

```diff
@@ -184,11 +204,117 @@ export class DB {
         return new Collection<T>(collectionName, this.db, this.grpcClient)
     }
 
+    public transact(fn: (tx: Session) => void) {
+        let sessionVar: Session;
+        this.beginTransaction().then(session => {
+            // tx started
+            sessionVar = session;
+            try {
+                // invoke user code
+                fn(session)
+                // user code successful
+                return session.commit()
+            } catch (error) {
+                // failed to run user code
+                if (sessionVar) {
```

Comment:
is it possible that at this point it is not resolved? If yes, then the user function is called with nil session and then even though they are expecting the call to be transactional it happened in non-transactional session?

Reply:
session is resolved at this point.

Reply:
then do we need this if check?